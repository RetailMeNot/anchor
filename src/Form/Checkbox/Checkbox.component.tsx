// REACT
import * as React from 'react';
const { forwardRef } = React;

// VENDOR
import styled, { css } from '@xstyled/styled-components';
import { compose } from '@xstyled/system';
import classnames from 'classnames';

// ANCHOR
import { Check } from '../..';
import { StyledIcon } from '../../Icon/utils';
import {
    space as spaceStyles,
    SpaceProps,
    color as colorStyles,
    ColorProps,
    backgroundColor as backgroundColorStyles,
    BackgroundColorProps,
} from '../../theme/system.theme';

const HiddenInput = styled('input')<CheckboxProps>`
    visibility: hidden;
    opacity: 0;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
`;

const StyledCheckbox = styled('label')<CheckboxProps>`
    box-sizing: border-box;
    position: relative;
    display: inline-flex;
    align-items: center;
    justify-content: center;

    user-select: none;
    border: dark;
    border-radius: 2px;

    ${compose(
        spaceStyles,
        colorStyles,
        backgroundColorStyles
    )}

    ${({ size, disabled }) =>
        css({
            height: size,
            width: size,
            cursor: disabled ? 'default' : 'pointer',
            opacity: disabled ? 0.4 : 1,
        })}

    ${StyledIcon} {
        transition: opacity 100ms ease-in-out;
        ${({ checked }) =>
            css({
                opacity: checked ? 1 : 0,
            })}
    }
`;

interface CheckboxProps extends SpaceProps, ColorProps, BackgroundColorProps {
    className?: string;
    id?: string;
    htmlFor?: string;

    checked?: boolean;
    disabled?: boolean;
    name?: string;

    size?: string;
    color?: string;

    inputProps?: any;
    onChange?: (arg: any) => any;
}

export const Checkbox = forwardRef(
    (
        {
            id,
            className,
            inputProps,
            color = 'primary.base',
            backgroundColor,
            checked,
            size = '1.375rem',
            disabled,
            onChange,
            ...props
        }: CheckboxProps,
        ref: React.RefObject<any>
    ): React.ReactElement<CheckboxProps> => (
        <StyledCheckbox
            className={classnames('anchor-checkbox', className)}
            disabled={disabled}
            htmlFor={id}
            color={color}
            backgroundColor={
                backgroundColor || (disabled ? 'neutrals.silver.base' : 'white')
            }
            size={size}
            checked={checked}
            {...props}
        >
            <Check scale="sm" />
            <HiddenInput
                type="checkbox"
                id={id}
                onChange={onChange}
                checked={checked}
                disabled={disabled}
                ref={ref}
                {...props}
                {...inputProps}
            />
        </StyledCheckbox>
    )
);
