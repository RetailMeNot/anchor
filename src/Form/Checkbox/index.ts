export { Checkbox } from './Checkbox.component';
