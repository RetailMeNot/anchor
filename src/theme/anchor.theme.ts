// ANCHOR THEME
import { radii } from './sizes.theme';
import { MODAL_THEME_KEY, MODAL_THEME } from '../Modal';

// Copied from Adapter.StandardBreakpoints mins
// todo: dedup- they should use a single source
const breakpoints = {
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
};

interface AnchorTheme {
    colors?: any;
    breakpoints?: any;
    radii?: any;
    [key: string]: any;
}

export const anchorTheme: AnchorTheme = {
    breakpoints,
    radii,
    [MODAL_THEME_KEY]: MODAL_THEME,
};
