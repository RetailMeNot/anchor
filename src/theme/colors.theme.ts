export default {
  neutrals: {
    charcoal: {
      base: '#323232',
      light: '#595959',
      dark: '#000000',
    },
    ash: {
      base: '#A6A6A6',
      light: '#D3D3D3',
      dark: '#808080',
    },
    silver: {
      base: '#F1F1F1',
      light: '#FAFAFA',
      dark: '#E7E7E7',
    },
    lavenderWhispers: {
      base: '#F7F5FC',
      light: '#FBFAFF',
      dark: '#F1EEFA',
    },
    white: {
      base: '#FFFFFF',
    },
  },
  main: {
    grapePurchase: {
      base: '#602D6C',
      light: '#91609C',
      dark: '#4B2354',
    },
    rx: {
      base: '#F23466',
      light: '#FF6B9C',
      dark: '#CC2D52',
    },
    genie: {
      base: '#4C18F3',
      light: '#7855D9',
      dark: '#42199A',
    },
  },
  accent: {
    savvyCyan: {
      base: '#0998D6',
      light: '#66CCFF',
      dark: '#0074A6',
    },
    tealBreaker: {
      base: '#00CCCD',
      light: '#00E0D1',
      dark: '#00BDBD',
    },
  },
  secondary: {
    dealEnvy: {
      base: '#14A382',
      light: '#00CDAD',
      dark: '#268068',
    },
    flashPink: {
      base: '#DF206C',
      light: '#E665A1',
      dark: '#B81D5B',
    },
    cyberMango: {
      base: '#EB663D',
      light: '#FA9F5A',
      dark: '#C74E30',
    },
    goldMine: {
      base: '#F5A623',
      light: '#FFC761',
      dark: '#CC861D',
    },
    fireSale: {
      base: '#EA4545',
      light: '#F16667',
      dark: '#CC3B3B',
    },
  },
  alert: {
    error: {
      base: '#CC3B3B',
    },
    success: {
      base: '#268068',
    },
  },
  rx: {
    purple: {
      base: '#A678B5',
    },
  },
};
