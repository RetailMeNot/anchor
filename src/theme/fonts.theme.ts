// VARIABLES
import variables from './variables.theme';

const { baseFontSize, fontFamily } = variables;

export default {
  baseFontSize,
  fontFamily,
  paragraph: {
    base: {
      fontSize: `${baseFontSize}px`,
      lineHeight: `${baseFontSize * 1.5}px`,
      fontWeight: 'normal',
    },
    informational: {
      fontSize: `${baseFontSize *  0.875}px`,
      lineHeight: `${baseFontSize * 1.125}px`,
      fontWeight: 'normal',
    },
  },
  navigation: {
    base: {
      fontSize: `${baseFontSize * .75}px`,
      lineHeight: `${baseFontSize}px`,
      fontWeight: '500',
    },
    menu: {
      fontSize: `${baseFontSize * .875}px`,
      lineHeight: `${baseFontSize}px`,
      small: {
        base: {
          fontSize: `${baseFontSize * 0.75}px`,
          fontWeight: '500',
        },
        active: {
          fontWeight: '700',
        },
      },
      large: {
        base: {
          fontWeight: '600',
        },
        active: {
          fontWeight: '700',
        },
      },
    },
  },
  heading: {
    h4: {
      fontSize: `${baseFontSize * 1.5}px`,
      lineHeight: `${baseFontSize * 2}px`,
      fontWeight: 'normal',
    },
    h3: {
      fontSize: `${baseFontSize * 1.75}px`,
      lineHeight: `${baseFontSize * 2}px`,
      fontWeight: 'normal',
    },
    h2: {
      fontSize: `${baseFontSize * 2}px`,
      lineHeight: `${baseFontSize * 2.5}px`,
      fontWeight: 'normal',
    },
    h1: {
      fontSize: `${baseFontSize * 2.25}px`,
      lineHeight: `${baseFontSize * 2.5}px`,
      fontWeight: 'normal',
    },
  },
  typography: {
    caption: {
      fontSize: `${baseFontSize * 0.75}px`,
      lineHeight: `${baseFontSize}px`,
      fontWeight: '500',
    },
    ui: {
      fontSize: `${baseFontSize * 0.875}px`,
      lineHeight: `${baseFontSize * 1.125}px`,
      fontWeight: 'normal',
    },
    body: {
      fontSize: `${baseFontSize}px`,
      lineHeight: `${baseFontSize * 1.5}px`,
      fontWeight: 'normal',
    },
    subHeading: {
      fontSize: `${baseFontSize * 1.125}px`,
      lineHeight: `${baseFontSize * 1.5}px`,
      fontWeight: 'normal',
    },
    subHeadingLarge: {
      fontSize: `${baseFontSize * 1.25}px`,
      lineHeight: `${baseFontSize * 1.5}px`,
      fontWeight: 'normal',
    },
    header: {
      small: {
        fontSize: `${baseFontSize * 1.5}px`,
        lineHeight: `${baseFontSize * 2}px`,
        fontWeight: 'normal',
      },
      medium: {
        fontSize: `${baseFontSize * 1.75}px`,
        lineHeight: `${baseFontSize * 2}px`,
        fontWeight: 'normal',
      },
      large: {
        fontSize: `${baseFontSize * 2}px`,
        lineHeight: `${baseFontSize * 2.5}px`,
        fontWeight: 'normal',
      },
      extraLarge: {
        fontSize: `${baseFontSize * 2.25}px`,
        lineHeight: `${baseFontSize * 2.5}px`,
        fontWeight: 'normal',
      },
    },
    display: {
      small: {
        fontSize: `${baseFontSize * 2.5}px`,
        lineHeight: `${baseFontSize * 3}px`,
        fontWeight: 'normal',
      },
      medium: {
        fontSize: `${baseFontSize * 3}px`,
        lineHeight: `${baseFontSize * 3.25}px`,
        fontWeight: 'normal',
      },
      large: {
        fontSize: `${baseFontSize * 3.5}px`,
        lineHeight: `${baseFontSize * 4}px`,
        fontWeight: 'normal',
      },
      extraLarge: {
        fontSize: `${baseFontSize * 4}px`,
        lineHeight: `${baseFontSize * 4.5}px`,
        fontWeight: 'normal',
      },
    },
  },
};
