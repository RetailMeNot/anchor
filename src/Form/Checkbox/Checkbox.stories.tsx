// REACT
import * as React from 'react';
// STORYBOOK
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';
import { ThemeProvider } from '@xstyled/styled-components';
// ANCHOR
import { RootTheme } from '../../theme/index';
// SUBJECT
import * as README from './README.md';
import { Checkbox } from './Checkbox.component';

const { useState } = React;

storiesOf('Components/Form/Checkbox', module)
    .addParameters({ readme: { sidebar: README } })
    .add('Default', () =>
        React.createElement(() => {
            const [checked, setChecked] = useState<boolean>(true);
            return (
                <ThemeProvider theme={RootTheme}>
                    <Checkbox
                        checked={checked}
                        disabled={boolean('disabled', false)}
                        onChange={() => setChecked(!checked)}
                    />
                </ThemeProvider>
            );
        })
    );
