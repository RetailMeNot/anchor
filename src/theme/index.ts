// Theme Files
import { typography } from './typography.theme';
import { ColorsTheme, colors } from './colors.theme';

// TODO: ============================================================ move all of these v consts to a different location

// When re-exporting an interface its type must be explicitly defined
import { Color } from './colors.theme';
export type Color = Color;

export { colors } from './colors.theme';
export { fonts } from './fonts.theme';
// I'm sorry internet, but this works.
import { sizes as Sizes } from './sizes.theme';
export const sizes = Sizes;

export { NormalizeCSS, GlobalCSS } from './GlobalStyles';
// TODO: ============================================================ move all of these ^ consts to a different location

import { BUTTON_KEY, BUTTON_THEME } from '../Button/Button.component';

export const RootTheme = {
    typography,
    radii: { none: '0', base: '4px', modal: '8px', circular: '1000px' },
    [BUTTON_KEY]: BUTTON_THEME,
    colors: ColorsTheme,
    breakpoints: {
        xs: 0,
    },
    // todo: we should likely remove the color.borders colors and just use this
    // or find a way to reference the colors.border colors here
    // e.g. `thin solid borders.base`
    borders: {
        light: `thin solid ${colors.silver.base}`,
        base: `thin solid ${colors.ash.light}`,
        dark: `thin solid ${colors.ash.dark}`,
    },
};
