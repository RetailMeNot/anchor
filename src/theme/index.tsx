// Theme Files
import { typography } from './typography.theme';
import { ColorsTheme } from './colors.theme';
import { ThemeProvider as XstyledThemeProvider } from '@xstyled/styled-components';
import styled from '@xstyled/styled-components';
import { space as spaceStyles, SpaceProps } from '@xstyled/system';
import { BUTTON_KEY, BUTTON_THEME } from '../Button';
import { INPUT_KEY, INPUT_THEME } from '../Form/Input/utils';
import React from 'react';
import {Button} from '../Button';
import classNames from 'classnames';

export const RootTheme = {
    typography,
    radii: { none: '0', base: '4px', modal: '8px', circular: '1000px' },
    [BUTTON_KEY]: BUTTON_THEME,
    [INPUT_KEY]: INPUT_THEME,
    colors: ColorsTheme,
    fonts: {
        base:
            'Avenir Next, Segoe UI, Helvetica Neue, Helvetica, Roboto, sans-serif',
        mono: `'SFMono-Regular', Consolas, Liberation Mono, Menlo, Courier, monospace`,
    },
    // These are all minimums
    breakpoints: {
        xs: 0,
        sm: 576,
        md: 768,
        lg: 992,
        xl: 1200,
        xxl: 1600,
    },
    // todo: we should likely remove the color.borders colors and just use this
    // or find a way to reference the colors.border colors here
    // e.g. `thin solid borders.base`
    borders: {
        light: `thin solid ${ColorsTheme.borders.light}`,
        base: `thin solid ${ColorsTheme.borders.base}`,
        dark: `thin solid ${ColorsTheme.borders.dark}`,
    },

    skeleton: {
        colorStart: '#E7E7E7',
        colorEnd: '#D3D3D3',
    },
};

// Exporting out @xstyled's ThemeProvider. Only way I could get it to work w/o issues.
// export class ThemeProvider2 extends XstyledThemeProvider {}

interface IterableObject {
    [s: number]: number | string | boolean | IterableObject;
  }

interface DebugProps {
    property: number | string | boolean | IterableObject;
    propertyName: string;
    rootProperty?: boolean;
}

interface StyledDebugProps {
    indent?: number;
}

const StyledDebug = styled('div')<StyledDebugProps>`
    font-family: mono;
    margin-top: 0.5rem;
    color: white;

    .category .category h2 {
        background-color: #ec8947;
    }

    .category .category .box {
        border-color: #ec8947;
    }

    .category .category .category h2 {
        background-color: #b84343;
        color: white;
    }

    .category .category .category .box {
        border-color: #b84343;
    }

    .category .category .category .category h2 {
        background-color: #91cdd2;
        color: black;
    }

    .category .category .category .category .box {
        border-color: #91cdd2;
    }

    .category .category .category .category .category h2 {
        background-color: #567224;
        color: white;
    }

    .category .category .category .category .category .box {
        border-color: #567224;
    }
`;

const StyledH2 = styled('h2')`
    ${spaceStyles}
    background-color: #ffea92;
    color: black;
    margin-bottom: 0;
    margin-top: 0;
    padding: 0.25rem 0.5rem 0.25rem 1rem;
    font-size: 1rem;
    cursor: pointer;
    border-radius: 0 1rem 1rem 0;
`;

interface BoxProps extends SpaceProps {
    show?: boolean;
}

const Box = styled('div')<BoxProps>`
    ${spaceStyles}
    width: 20rem;
    border-left: solid #ffea92 2px;
    margin: 0.5rem;
    ${({show}) => `display: ${show ? 'block' : 'none'}`}
`;

const Category = styled('div')<SpaceProps>`
    ${spaceStyles}
    width: 20rem;
`;


const ColorBox = styled('div')<any>`
    width: 1rem;
    height: 1rem;
    display: inline-block;
    vertical-align: middle;
    margin-left: 0.5rem;
    background-color: ${({backgroundColor}) => backgroundColor};
    border: solid white 1px;
`;

const Debug = ({property, propertyName}: DebugProps) => {
    const [show, setShow] = React.useState(false);

    return(
        <StyledDebug>
            {(property || property === 0)
                ? (
                    typeof property === 'number' ||
                    typeof property === 'string' ||
                    typeof property === 'boolean' ||
                    typeof property === 'function'
                        ? (
                                <>
                                    <span style={{fontWeight: 'bold'}}>{propertyName}: </span>
                                    <span>
                                        {typeof property === 'function'
                                            ? 'function'
                                            : /^#[a-zA-Z0-9]{4,7}/.test(property.toString())
                                                ? (<>
                                                    <span style={{fontStyle: 'italic'}}>{property.toString()}</span>
                                                    <ColorBox backgroundColor={property} />
                                                    </>)
                                                : <span style={{fontStyle: 'italic'}}>{property.toString()}</span>
                                        }
                                    </span>
                                </>
                            )
                        :
                            <Category pl={0} className="category">
                                <StyledH2 onClick={() => setShow(!show)}>{propertyName}</StyledH2>
                                <Box pl={2} className="box" show={show}>
                                    {Object.values(property).map((_property, index) => (
                                        <Debug
                                            key={index}
                                            property={_property}
                                            propertyName={Object.getOwnPropertyNames(property)[index]}
                                        />
                                    ))}
                                </Box>
                            </Category>
                    )
                : (
                    <>
                       Empty.
                    </>
                )
            }
        </StyledDebug>
    );
};

const DebugWrapper = styled('div')`
    background-color: rgba(0,0,0,.75);
    border-left-color: transparent;
    position: fixed;
    left: 0;
    top: 0;
    height: 100%;
    overflow: hidden;
    width: 30rem;
    padding: 1rem 2rem 1rem 0;

    &.closed {
        height:2.4rem;
        width: auto;
        padding-right: 1rem;

        .debug-scroll {
            display: none;
        }
    }
`;

const DebugScroll = styled('div')`
    height: 100%;
    overflow-y: auto;
    width: 100%;
`;

interface ThemeProviderProps {
    children?: any;
    debug?: boolean;
    theme?: any;
}

export const ThemeProvider = ({ children, debug, theme }: ThemeProviderProps) => {
    const isOpen = localStorage.themeProviderDebug === 'true';
    const [isOpenState, setIsOpenState] = React.useState(isOpen);

    return (
        <XstyledThemeProvider theme={theme}>
            {debug &&
                <DebugWrapper className={classNames(!isOpenState ? 'closed' : null)}>
                    <Button
                        onClick={() => {
                            localStorage.themeProviderDebug = !isOpenState;
                            setIsOpenState(!isOpenState);
                        }}
                        size="sm"
                    >
                        ThemeProvider Debug
                    </Button>

                    <DebugScroll className="debug-scroll">
                        <Debug property={theme} propertyName="Theme" />
                    </DebugScroll>
                </DebugWrapper>
            }
            {children}
        </XstyledThemeProvider>
    );
};
