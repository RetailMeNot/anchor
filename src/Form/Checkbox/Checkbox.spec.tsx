// VENDOR
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ThemeProvider } from '@xstyled/styled-components';
// ANCHOR
import { RootTheme } from '../../theme/index';
// COMPONENT
import { Checkbox } from './Checkbox.component';

describe('Component: Checkbox', () => {
    it('should be defined', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Checkbox />
            </ThemeProvider>
        );

        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should be disableable', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Checkbox disabled />
            </ThemeProvider>
        );

        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('can be checked', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <Checkbox checked />
            </ThemeProvider>
        );

        const tree = renderer.create(subject).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
