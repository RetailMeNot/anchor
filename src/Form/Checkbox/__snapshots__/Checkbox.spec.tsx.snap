// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Checkbox can be checked 1`] = `
.c2 {
  display: inline-block;
  height: 0.875rem;
  width: 0.875rem;
  line-height: 0;
}

.c3 {
  visibility: hidden;
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c4 .c1 {
  -webkit-transition: opacity 100ms ease-in-out;
  transition: opacity 100ms ease-in-out;
  opacity: 0;
}

.c5 .c1 {
  -webkit-transition: opacity 100ms ease-in-out;
  transition: opacity 100ms ease-in-out;
  opacity: 0;
}

.c0 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: thin solid #808080;
  border-radius: 2px;
  color: #602D6C;
  background-color: white;
  height: 1.375rem;
  width: 1.375rem;
  cursor: pointer;
  opacity: 1;
}

.c0 .c1 {
  -webkit-transition: opacity 100ms ease-in-out;
  transition: opacity 100ms ease-in-out;
  opacity: 1;
}

<label
  checked={true}
  className="anchor-toggle c0"
  color="primary.base"
  size="1.375rem"
>
  <span
    className="anchor-icon check c1 c2"
    scale="sm"
  >
    <svg
      height={14}
      viewBox="0 0 16 16"
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h16v16H0z"
        />
        <path
          d="M14.627 5.133l-7.637 7.5a1.282 1.282 0 0 1-1.8 0l-3.817-3.75a1.235 1.235 0 0 1 0-1.768 1.288 1.288 0 0 1 1.799 0L6.09 9.982l6.737-6.617a1.291 1.291 0 0 1 1.8 0c.496.488.496 1.28 0 1.768"
          fill="currentColor"
        />
      </g>
    </svg>
  </span>
  <input
    checked={true}
    className="c3"
    type="checkbox"
  />
</label>
`;

exports[`Component: Checkbox should be defined 1`] = `
.c2 {
  display: inline-block;
  height: 0.875rem;
  width: 0.875rem;
  line-height: 0;
}

.c3 {
  visibility: hidden;
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c0 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: thin solid #808080;
  border-radius: 2px;
  color: #602D6C;
  background-color: white;
  height: 1.375rem;
  width: 1.375rem;
  cursor: pointer;
  opacity: 1;
}

.c0 .c1 {
  -webkit-transition: opacity 100ms ease-in-out;
  transition: opacity 100ms ease-in-out;
  opacity: 0;
}

<label
  className="anchor-toggle c0"
  color="primary.base"
  size="1.375rem"
>
  <span
    className="anchor-icon check c1 c2"
    scale="sm"
  >
    <svg
      height={14}
      viewBox="0 0 16 16"
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h16v16H0z"
        />
        <path
          d="M14.627 5.133l-7.637 7.5a1.282 1.282 0 0 1-1.8 0l-3.817-3.75a1.235 1.235 0 0 1 0-1.768 1.288 1.288 0 0 1 1.799 0L6.09 9.982l6.737-6.617a1.291 1.291 0 0 1 1.8 0c.496.488.496 1.28 0 1.768"
          fill="currentColor"
        />
      </g>
    </svg>
  </span>
  <input
    className="c3"
    type="checkbox"
  />
</label>
`;

exports[`Component: Checkbox should be disableable 1`] = `
.c2 {
  display: inline-block;
  height: 0.875rem;
  width: 0.875rem;
  line-height: 0;
}

.c3 {
  visibility: hidden;
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

.c4 .c1 {
  -webkit-transition: opacity 100ms ease-in-out;
  transition: opacity 100ms ease-in-out;
  opacity: 0;
}

.c0 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: thin solid #808080;
  border-radius: 2px;
  color: #602D6C;
  background-color: #F1F1F1;
  height: 1.375rem;
  width: 1.375rem;
  cursor: default;
  opacity: 0.4;
}

.c0 .c1 {
  -webkit-transition: opacity 100ms ease-in-out;
  transition: opacity 100ms ease-in-out;
  opacity: 0;
}

<label
  className="anchor-toggle c0"
  color="primary.base"
  disabled={true}
  size="1.375rem"
>
  <span
    className="anchor-icon check c1 c2"
    scale="sm"
  >
    <svg
      height={14}
      viewBox="0 0 16 16"
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h16v16H0z"
        />
        <path
          d="M14.627 5.133l-7.637 7.5a1.282 1.282 0 0 1-1.8 0l-3.817-3.75a1.235 1.235 0 0 1 0-1.768 1.288 1.288 0 0 1 1.799 0L6.09 9.982l6.737-6.617a1.291 1.291 0 0 1 1.8 0c.496.488.496 1.28 0 1.768"
          fill="currentColor"
        />
      </g>
    </svg>
  </span>
  <input
    className="c3"
    disabled={true}
    type="checkbox"
  />
</label>
`;
